packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -15,14 +15,27 @@
  */
 import path from 'path';
 import express from 'express';
-import aws from 'aws-sdk';
+import { PutObjectCommandOutput, S3 } from '@aws-sdk/client-s3';
 import { Logger } from 'winston';
 import { Entity, EntityName } from '@backstage/catalog-model';
 import { Config } from '@backstage/config';
 import { getHeadersForFileExtension, getFileTreeRecursively } from './helpers';
 import { PublisherBase, PublishRequest } from './types';
-import { ManagedUpload } from 'aws-sdk/clients/s3';
 import fs from 'fs-extra';
+import { Readable } from 'stream';
+
+const streamToString = (stream: Readable): Promise<string> => {
+  try {
```

Comment:
This can never throw. The Promise constructor is the only thing being invoked here. The actual function inside is run "inside" the promise, and will only lead to it being rejected. If I understand things correctly, you want this try/catch inside the promise instead.